app/src/main/java/mainburg/planetenweg/LocationMarker.java

```diff
@@ -0,0 +1,146 @@
+                == PackageManager.PERMISSION_GRANTED) {
+            refresh();
+        } else {
+            ActivityCompat.requestPermissions(activity, new String[] {
+                    android.Manifest.permission.ACCESS_FINE_LOCATION
+            },GPS_PERMISSION_REQUEST_CODE);
+        }
+    }
+    /**
+     * Disables the Location Marker
+     */
+    public void disable() {
+        position.remove();
+        active = false;
+    }
+
+    public boolean isEnabled() {
+        return active;
+    }
+
+    /**
+     * Sends a new request to determine the user's location.
+     * Make sure you have the required GPS permissions before calling this method.
+     */
+    public void refresh() {
+        gps.requestLocationUpdates(
+                LocationManager.GPS_PROVIDER,
+                5000, //the minimum time interval for notifications, in milliseconds. This field is only used as a hint
+                //to conserve power, and actual time between location, updates may be greater or lesser than this value. (See https://androidexample.com/GPS_Basic__-__Android_Example/index.php?view=article_discription&aid=68&aaid=93)
+                5,//the minimum distance interval for notifications
+                this);//The listener to which to send the updates
+
+        locationFound = false;
+        Timer timer = new Timer();
+        timer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                if (!locationFound) {
+                    Toast.makeText(activity, "Position konnte nicht ermittelt werden.", Toast.LENGTH_LONG);
+                }
+            }
+        }, 10000);
+
+    }
+
+    /************* Called after each 10 sec **********/
+    @Override
+    public void onLocationChanged(Location location) {
+        if (!active) {
+            return;
+        }
+        locationFound = true;
+        if (position != null) {
+            position.remove();
+        }
+
+        LatLng loc = new LatLng(location.getLatitude(), location.getLongitude());
+        position = map.addMarker(new MarkerOptions()
+                .position(loc)
+                .title("Sie sind hier."));
+
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(loc, map.getCameraPosition().zoom));
+    }
+
+    @Override
+    public void onProviderDisabled(String provider) {
+
+        /******** Called when User disables Gps *********/
+        Toast.makeText(activity.getBaseContext(), "GPS deaktiviert", Toast.LENGTH_LONG).show();
+
+        if (position != null) {
+            position.remove();
+        }
+    }
+
+    @Override
+    public void onProviderEnabled(String provider) {
+
+        /******** Called when User enables Gps  *********/
+
+        Toast.makeText(activity.getBaseContext(), "GPS aktiviert", Toast.LENGTH_LONG).show();
+    }
+
+    @Override
+    public void onStatusChanged(String provider, int status, Bundle extras) {
```

Comment:
Is this supposed to stay a no-op?

Reply:
This method is implementation of the interface. Still I don't know a use for it now.